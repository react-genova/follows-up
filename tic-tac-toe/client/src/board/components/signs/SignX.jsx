import React from 'react';
import PropTypes from 'prop-types';

const SignX = ({ ghost }) => (
    <svg width="100%" height="100%" viewBox="0 0 100 100">
        <line x1="0" y1="0" x2="100" y2="100" stroke={ghost ? 'lightgray' : 'cyan'} strokeWidth="8" />
        <line x1="0" y1="100" x2="100" y2="0" stroke={ghost ? 'lightgray' : 'cyan'} strokeWidth="8" />
    </svg>
);

SignX.propTypes = {
    ghost: PropTypes.bool,
};

SignX.defaultProps = {
    ghost: false,
};


export default SignX;
