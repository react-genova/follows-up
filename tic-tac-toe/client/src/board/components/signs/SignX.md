### On light background

```jsx
<div style={{ display:"flex", width:"100px", height:"50px", padding:"2px" }}>
    <SignX />
    <SignX ghost />
</div>
```

### On dark background

```jsx
<div style={{ display:"flex", width:"100px", height:"50px", padding:"2px", backgroundColor: "rgb(50,50,50)" }}>
    <SignX />
    <SignX ghost />
</div>
```

### A little bigger

```jsx
<div style={{ display:"flex", width:"200px", height:"100px", padding:"5px", backgroundColor: "rgb(50,50,50)" }}>
    <SignX />
    <SignX ghost />
</div>
```