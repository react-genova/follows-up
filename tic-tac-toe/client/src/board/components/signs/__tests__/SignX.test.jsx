import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';
import SignX from '../SignX';

describe('SignX component', () => {
    afterEach(cleanup);

    it('renders without crashing', () => {
        const { container } = render(<SignX />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders without crashing (ghost)', () => {
        const { container } = render(<SignX ghost />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
