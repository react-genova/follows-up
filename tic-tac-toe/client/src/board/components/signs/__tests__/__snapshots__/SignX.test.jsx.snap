// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignX component renders without crashing (ghost) 1`] = `
<svg
  class="sign-x"
  height="100%"
  viewBox="0 0 100 100"
  width="100%"
>
  <line
    stroke="lightgray"
    stroke-width="8"
    x1="0"
    x2="100"
    y1="0"
    y2="100"
  />
  <line
    stroke="lightgray"
    stroke-width="8"
    x1="0"
    x2="100"
    y1="100"
    y2="0"
  />
</svg>
`;

exports[`SignX component renders without crashing 1`] = `
<svg
  class="sign-x"
  height="100%"
  viewBox="0 0 100 100"
  width="100%"
>
  <line
    stroke="cyan"
    stroke-width="8"
    x1="0"
    x2="100"
    y1="0"
    y2="100"
  />
  <line
    stroke="cyan"
    stroke-width="8"
    x1="0"
    x2="100"
    y1="100"
    y2="0"
  />
</svg>
`;
